velocity/src/main/java/net/william278/huskchat/velocity/HuskChatVelocity.java

```diff
@@ -94,6 +95,10 @@ public void onProxyInitialization(ProxyInitializeEvent event) {
         // Load messages
         reloadMessages();
 
+        // Load saved social spy state
+        PlayerCache.setDataFolder(getDataFolder());
+        PlayerCache.loadSpy();
```

Comment:
Have loadSpy and removeSpy `throw IOException` and catch here / also in the bungee file. You can then properly handle the logging by getting the logger with the exception; `getLoggingAdapter().log(...)`